client/asset/btc/spv.go

```diff
@@ -1244,6 +1241,55 @@ func (w *spvWallet) startWallet() error {
 	return nil
 }
 
+// deleteWalletData will delete all files except peers.json from the
+// wallet's directory.
+func (w *spvWallet) deleteWalletData() error {
+	contents, err := os.ReadDir(w.netDir)
+	if err != nil {
+		return err
+	}
+
+	for _, f := range contents {
+		if f.Name() == "peers.json" {
+			continue
+		}
+		path := filepath.Join(w.netDir, f.Name())
+		if f.IsDir() {
+			_ = os.RemoveAll(path)
+		} else {
+			_ = os.Remove(path)
+		}
+	}
+
+	return nil
+}
+
+// numDerivedAddresses returns the number of internal and external addresses
+// that the wallet has derived.
+func (w *spvWallet) numDerivedAddresses() (internal, external uint32, err error) {
+	btcw, ok := w.loader.LoadedWallet()
+	if !ok {
+		return 0, 0, err
+	}
+
+	scopedKeyManager, err := btcw.Manager.FetchScopedKeyManager(waddrmgr.KeyScopeBIP0084)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	wdb := btcw.Database()
+
+	err = walletdb.View(wdb, func(dbtx walletdb.ReadTx) error {
+		ns := dbtx.ReadBucket([]byte("waddrmgr"))
+		props, err := scopedKeyManager.AccountProperties(ns, w.acctNum)
+		internal = props.InternalKeyCount
+		external = props.ExternalKeyCount
+		return err
```

Comment:
Should return directly after the error if it is not nil. Otherwise props.Something may panic if props is a pointer. At the end you can return `nil` to satisfy the function return.